File: 06_dart/01_two_sum.dart
void main() {
  runTest([2, 7, 11, 15], 9);
  runTest([3, 2, 4], 6);
  runTest([3, 3], 6);
}

List<int> twoSum(List<int> nums, int target) {
  final Map<int, int> seen = {};

  for (int i = 0; i < nums.length; i++) {
    int complement = target - nums[i];
    if (seen.containsKey(complement)) {
      return [seen[complement]!, i];
    }
    seen[nums[i]] = i;
  }
  return []; // unreachable as per problem guarantees
}

void runTest(List<int> nums, int target) {
  final result = twoSum(nums, target);
  print('nums=$nums target=$target --> result=$result');
}

File: 07_kotlin/01_two_sum.kt

fun twoSum(nums: IntArray, target: Int): IntArray{
    val seen = HashMap<Int, Int>()

    for(i in nums.indices){
        val complement = target - nums[i]

        if (seen.containsKey(complement)){
            return intArrayOf(seen[complement]!!, i)
        }

        seen[nums[i]] = i
    }
    return intArrayOf() // unreachable as per problem
}


fun runTest(nums: IntArray, target: Int){

    val result = twoSum(nums, target)

    println("nums=${nums.contentToString()} target=$target --> result=${result.contentToString()}")
}

fun main(args: Array<String>) {
    // Sample LeetCode tests
    runTest(intArrayOf(2, 7, 11, 15), 9)
    runTest(intArrayOf(3, 2, 4), 6)
    runTest(intArrayOf(3, 3), 6)
}